forge/test/operators/utils/utils.py

```diff
@@ -0,0 +1,179 @@
+from typing import Optional, List, Dict, Type, Union
+
+from forge import ForgeModule, Module, VerifyConfig
+from forge.op_repo import TensorShape
+from forge.verify import TestKind  #, verify_module
+from forge.config import _get_global_compiler_config
+from forge._C import MathFidelity
+
+from .compat import TestDevice, verify_module
+
+
+# All supported framework model types
+FrameworkModelType = Union [
+    Type[torch.nn.Module],
+]
+
+
+class ShapeUtils:
+
+    @staticmethod
+    def reduce_microbatch_size(shape: TensorShape) -> TensorShape:
+        '''
+        Reduce microbatch dimension of a shape to 1
+        Usually used for calculating shape of a constant tensor
+        '''
+        return (1, ) + shape[1:]
+    
+    @staticmethod
+    def extend_shapes_with_id(shapes: List[TensorShape]) -> List[TensorShape]:
+        '''
+        Extend shapes with an id
+        '''
+        shapes_with_ids = list()
+        for shape in shapes:
+            if type(shape) is tuple:
+                shapes_with_ids.append(pytest.param(shape, marks=tuple(), id=f"shape={shape}"))
+            else:
+                shapes_with_ids.append(pytest.param(shape.values[0], marks=shape.marks, id=f"shape={shape.values[0]}"))
+        return shapes_with_ids
+
+
+@dataclass(frozen=True)
+class InputSourceFlag:
+    '''Dataclass for specifying compiler flags for specific input source'''
+    input_queues_on_host: bool
+    set_default_dram_parameters: bool
+    default_dram_parameters: Optional[bool]
+
+
+class InputSourceFlags(Enum):
+    '''Enums defining input source flags'''
+    FROM_HOST = InputSourceFlag(True, False, None)
+    FROM_DRAM = InputSourceFlag(False, False, None)
+    FROM_DRAM_PROLOGUED = InputSourceFlag(False, True, False)
+    FROM_DRAM_NOT_PROLOGUED = InputSourceFlag(False, True, True)
+    FROM_DRAM_PROLOGUE_MICROBATCH_SIZE = InputSourceFlag(False, True, None)
+
+
+class CompilerUtils:
+    '''Utility functions for Forge compiler configuration'''
+
+    @staticmethod
+    def set_input_source(input_source_flag: InputSourceFlag):
+        '''Set compiler configuration for input source'''
+        compiler_cfg = _get_global_compiler_config()
+        compiler_cfg.input_queues_on_host = input_source_flag.input_queues_on_host
+        if input_source_flag.set_default_dram_parameters:
+            compiler_cfg.default_dram_parameters = input_source_flag.default_dram_parameters
+
+    @staticmethod
+    def set_math_fidelity(math_fidelity: MathFidelity):
+        '''Set compiler configuration for math fidelity'''
+        compiler_cfg = _get_global_compiler_config()
+        compiler_cfg.default_math_fidelity = math_fidelity
+
+
+class VerifyUtils:
+    '''Utility functions for Forge verification'''
+
+    @staticmethod
+    def verify(
+        model: Module,
+        test_device: TestDevice,
+        input_shapes: List[TensorShape],
+        input_params: List[Dict] = [],
+        check_pcc: Optional[bool] = True,
```

Comment:
Do we need a check_pcc flag? Is there a case when we don't want to check PCC?

Reply:
Probably there is not need to support skipping pcc check. Removed the option